lib/iterator.js

```diff
@@ -159,6 +159,14 @@ class Iterator {
   enumerate() {
     return new EnumerateIterator(this);
   }
+
+  join(sep = ',', prefix = '', suffix = '') {
+    let result = prefix;
+    for (const value of this) {
+      result += value + sep;
+    }
+    return result.slice(0, -sep.length) + suffix;
```

Comment:
Nit: Instead of using `slice()` here, I'd prefer to have a separate boolean variable that will determine the addition of the separator, thus avoiding the addition of the separator after the last value. For example:
```js
    let first = true;
    for (const value of this) {
      if (!first) result += sep;
      result += value;
      first = false;
    }
    return result + suffix;
```

Reply:
What about:
```js
let [done, first] = this.next();
if (!done) {
  result += first;
  for (const value of this) {
    resulr += sep + value;
  }  
}
return result + sufix;
```